### PR TITLE
chore(website): add info for java options

### PR DESCRIPTION
**Pull request by SumantxD**
this PR tries to add info for the java specific options as mentioned in the issue #1392

### CODE REVIEW

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx [hunk 1]
```diff
@@ -143,8 +169,15 @@ class JavaGeneratorOptions extends React.Component<
             />
           </label>
         </li>
-        <li>
-          <label className="flex items-center py-2 justify-between cursor-pointer">
+        <li className=' flex items-center'>
+          <InfoModal text="Java array type :">
+            <p>
+            This option allows you to swithch between rendering collections as List type or Array.
```
Comment (jonaslagoni) [inline modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx:175]:
```suggestion
            This option allows you to switch between rendering collections as List type or Array.
```

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx [hunk 2]
```diff
@@ -143,8 +169,15 @@ class JavaGeneratorOptions extends React.Component<
...
+            <br/><br/>
+            the default value is Array.
```
Comment (jonaslagoni) [inline modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx:177]:
```suggestion
            The default value is Array.
```

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx [hunk 3]
```diff
@@ -187,8 +234,15 @@ class JavaGeneratorOptions extends React.Component<
...
           </label>
         </li>
-        <li>
-          <label className="flex items-center py-2 justify-between cursor-pointer">
+        <li className=' flex items-center'>
+          <InfoModal text="Include Overwrite toString Support :">
+            <p>
+            In Java, the toString() method is used to provide a string representation of an object. The default implementation in the Object class returns a string consisting of the class name, an "@" symbol, and the hexadecimal representation of the object's hash code.
+            <br/><br/>
+            By enabling the option to overwrite toString() support, you can define your own implementation of the toString() method for a class. This allows you to provide a more meaningful and customized string representation of the object based on its properties.
```
Comment (jonaslagoni) [inline modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx:242]: This is very generic wording about what `toString` is, however it does not explain what it does to the code generation.

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx [hunk 4]
```diff
@@ -173,8 +213,15 @@ class JavaGeneratorOptions extends React.Component<
...
           </label>
         </li>
-        <li>
-          <label className="flex items-center py-2 justify-between cursor-pointer">
+        <li className=' flex items-center'>
+          <InfoModal text="Include Overwrite Equal Support :">
+            <p>
+            n Java, the equals() method is used to determine if two objects are equal based on their content rather than their memory addresses. The default implementation of equals() in the Object class performs a reference equality check, meaning it only returns true if the compared objects are the same instance in memory.
+            <br/><br/>
+            By enabling the option to overwrite equal support, you can provide your own implementation of the equals() method in a class. This allows you to define the equality criteria based on the specific properties and requirements of the class.
```
Comment (jonaslagoni) [inline modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx:221]: This is very generic wording about what `toString` is, however it does not explain what it does to the code generation.

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx [hunk 5]
```diff
@@ -201,8 +255,15 @@ class JavaGeneratorOptions extends React.Component<
             />
           </label>
         </li>
-        <li>
-          <label className="flex items-center py-2 justify-between cursor-pointer">
+        <li className=' flex items-center'>
+          <InfoModal text="Include javaDocs :">
+            <p>
+            Enabling this option will incude the description of the properties as comments in the generated code.
```
Comment (jonaslagoni) [inline modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx:261]:
```suggestion
            Enabling this option will include the description of the properties as comments in the generated code.
```

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx [hunk 6]
```diff
@@ -159,8 +192,15 @@ class JavaGeneratorOptions extends React.Component<
...
           </label>
         </li>
-        <li>
-          <label className="flex items-center py-2 justify-between cursor-pointer">
+        <li className=' flex items-center'>
+          <InfoModal text="Include Overwrite HashCode Support :">
+            <p>
+            In Java, the hashCode() method is used to generate a unique numeric value (hash code) for an object. The default implementation of hashCode() in the Object class generates hash codes based on the memory address of the object, which may not be suitable for all classes.
+            <br/><br/>
+            By enabling the option to overwrite hashCode support, you can provide your own implementation of the hashCode() method in a class. This allows you to define a hash code generation algorithm based on the specific properties and requirements of the class.
```
Comment (jonaslagoni) [inline modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx:200]: This is very generic wording about what toString is, however it does not explain what it does to the code generation.

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx [hunk 7]
```diff
@@ -115,8 +123,15 @@ class JavaGeneratorOptions extends React.Component<
...
           </label>
         </li>
-        <li>
-          <label className="flex items-center py-2 justify-between cursor-pointer">
+        <li className=' flex items-center'>
+          <InfoModal text="Include Jackson serialization :">
+            <p>
+            Jackson annotations are a set of annotations provided by the Jackson library, which is a popular Java-based JSON processing library. These annotations are used to configure and control how Java objects are serialized to JSON and deserialized from JSON.
+            <br/><br/>
+            Annotations in Java are represented by the @ symbol followed by the annotation name.
```
Comment (jonaslagoni) [inline modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx:131]: This is very generic wording about what toString is, however it does not explain what it does to the code generation.

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Got a few comments 👍 Otherwise it looks good 😄

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx (hunk 1 above)
Comment (SumantxD) [reply modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx]: fixed!! :tada:

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx (hunk 3 above)
Comment (SumantxD) [reply modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx]: fixed!! :tada:

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx (hunk 2 above)
Comment (SumantxD) [reply modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx]: fixed!! :tada:

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx (hunk 5 above)
Comment (SumantxD) [reply modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx]: fixed!! :tada:

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx (hunk 7 above)
Comment (SumantxD) [reply modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx]: fixed!! :tada:

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx (hunk 6 above)
Comment (SumantxD) [reply modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx]: fixed!! :tada:

File: modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx (hunk 4 above)
Comment (SumantxD) [reply modelina-website/src/components/playground/options/JavaGeneratorOptions.tsx]: fixed!! :tada:

**Comment by SumantxD** [comment]
@jonaslagoni sorry for the delay, i was busy with a bunch of other stuffs and lost track of this issue :sweat_smile: 
I have tried fixing the requested changes, :smile:

**Review by jonaslagoni — APPROVED** [review]
Nice @SumantxD 👌

**Pull request merged**